internal/mithril-persistence/src/sqlite/transaction.rs

```diff
@@ -0,0 +1,174 @@
+use crate::sqlite::SqliteConnection;
+
+/// Sqlite transaction wrapper.
+///
+/// Transactions are automatically rolled back if this struct object is dropped and
+/// the transaction was not committed.
+pub struct Transaction<'a> {
+    connection: &'a SqliteConnection,
+    // An active transaction is one that has yet to be committed or rolled back.
+    is_active: bool,
+}
+
+impl<'a> Transaction<'a> {
+    /// Begin a new transaction.
+    pub fn begin(connection: &'a SqliteConnection) -> Result<Self, sqlite::Error> {
+        connection.execute("BEGIN TRANSACTION")?;
+        Ok(Self {
+            connection,
+            is_active: true,
+        })
+    }
+
+    /// Commit the transaction.
+    pub fn commit(mut self) -> Result<(), sqlite::Error> {
+        self.is_active = false;
+        self.connection.execute("COMMIT TRANSACTION")?;
```

Comment:
We can extract a function that execute and set is_active to false to reduce places where we modify this value.